extension/src/background/messageListener/handlers/addCollectible.ts

```diff
@@ -0,0 +1,48 @@
+import { AddCollectibleMessage } from "@shared/api/types/message-request";
+import { CollectibleContract } from "@shared/api/types/types";
+import { DataStorageAccess } from "background/helpers/dataStorageAccess";
+import { COLLECTIBLES_ID } from "constants/localStorageTypes";
+
+export const addCollectible = async ({
+  request,
+  localStore,
+}: {
+  request: AddCollectibleMessage;
+  localStore: DataStorageAccess;
+}) => {
+  const { network, publicKey, collectibleAddress, collectibleTokenId } =
+    request;
+
+  const collectibles = (await localStore.getItem(COLLECTIBLES_ID)) || {};
+  const networkCollectibles = collectibles[network] || {};
```

Comment:
when collectibles are saved, we delineate by both network and public key so (for ex) a collectible a user adds on G1 on testnet doesn't get loaded on G1 for mainnet. Similarly, a collectible added to G1 doesn't get loaded on G2